### DISCUSSION TITLE
brew install screen - does not compile

### DISCUSSION THREAD

**Discussion by sukucorp** [Homebrew/discussions#1651] [Everyday usage]
Hi,

Can someone using the standard prefix /usr/local confirm that screen formula can be installed on Big Sur.

I received the following error:
```
==> ./autogen.sh
Last 15 lines from /Users/admin/Library/Logs/Homebrew/screen/01.autogen.sh:
autoheader: warning: WARNING: Using auxiliary files such as 'acconfig.h', 'config.h.bot'
autoheader: WARNING: and 'config.h.top', to define templates for 'config.h.in'
autoheader: WARNING: is deprecated and discouraged.
autoheader: 
autoheader: WARNING: Using the third argument of 'AC_DEFINE_UNQUOTED' and
autoheader: WARNING: 'AC_DEFINE' allows one to define a template without
autoheader: WARNING: 'acconfig.h':
autoheader: 
autoheader: WARNING:   AC_DEFINE([NEED_FUNC_MAIN], 1,
autoheader: 		[Define if a function 'main' is needed.])
autoheader: 
autoheader: WARNING: More sophisticated templates can also be produced, see the
autoheader: WARNING: documentation.
configure.ac: error: required file 'install-sh' not found
configure.ac:   try running autoreconf --install

```
Replacing "./autogen.sh" with "autoreconf -i" or "autoreconf --install" fixed the issue.

**Comment by SMillerDev** [comment]
Works fine for me.

**Reply by sukucorp** [reply]
As indicated below, the installation will fail if --build-from-source is used. Three options:
1. Build dependency should be changed "autoconf@2.69" instead of "autoconf".
2. "./autogen.sh" should be changed to "autoreconf --install".
3. "./autogen.sh" should be removed since the package compiles cleanly without regenerating the autotool files.

**Comment by cho-m** [comment]
Installing Homebrew's existing `screen` bottle should work.

If you are trying to `--build-from-source` or use `--head`, then it may fail due to newer `autoconf` version.

---

All `autogen.sh` does is
```sh
#!/bin/sh
exec autoreconf
```

The problem seems to be related to newer `autoconf@2.71` vs. version used upstream.
Adding some extra flags like `autoreconf --install` can get help deal with these issues.
Alternatively using `autoconf@2.69` should match version Homebrew had previously bottled with.

**Reply by sukucorp** [reply]
Since I use a non-standard prefix, brew defaults to --build-from-source.

I have raised a ticket upstream to modify autogen.sh.
